jupyter_client/threaded.py

```diff
@@ -333,6 +336,19 @@ def _check_kernel_info_reply(self, msg: dict[str, Any]) -> None:
 
     def stop_channels(self) -> None:
         """Stop the channels on the client."""
+        # Close channel streams while ioloop is still running
+        # This must happen before stopping the ioloop thread, otherwise
+        # the ZMQ streams can't be properly unregistered from the event loop
+        if self.ioloop_thread and self.ioloop_thread.is_alive():
+            if self._shell_channel is not None:
+                self._shell_channel.close()
+            if self._iopub_channel is not None:
+                self._iopub_channel.close()
+            if self._stdin_channel is not None:
+                self._stdin_channel.close()
+            if self._control_channel is not None:
+                self._control_channel.close()
+
```

Comment:
@minrk we were running into an issue with QtConsole's tests where the channels not being closed before the ioloop thread was stopped would cause the ZMQ stream to have cleanup errors. A big change that triggered this is that the `IOLoopThread.run()` has a `finally: loop.close()` on it. If I close each channel first all is set.

Should `stop()` imply `close()` for threaded channels?